utilities/load-test/loadtest/loadtest_test.go

```diff
@@ -413,3 +413,95 @@ func TestAllApplication(t *testing.T) {
 	}
 	t.Log("Prometheus, along with 20 guestbook applications are created, deployed and deleted successfully!")
 }
+
+func TestNResource(t *testing.T) {
+	// TEST 1: Deply 100 guestbook in different namespace
+	t.Log("Testing for 100 applications that uses lightweight non-Pod-based resources\n")
+
+	t.Log("\n\nPods Memory Resource before creation:\n\n")
+	memoryInit := GetPodInfo(kubeconfig, masterUrl, namespace, podName)
+	PodInfoParse(memoryInit)
+
+	// Get Pod Restart Count
+	PodC := PodRestartCount(namespace)
+
+	for i := 1; i < 101; i++ {
+		applicationTest := &appv1.Application{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Application",
+				APIVersion: "argoproj.io/v1alpha1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "test-app-" + fmt.Sprintf("%d", i),
+				Namespace: "argocd",
+			},
+			Spec: appv1.ApplicationSpec{
```

Comment:
In order to ensure that Argo CD starts synchronizing the `Application` resource as soon as it is created, you can use `.spec.syncPolicy.automated`, like [this](https://github.com/argoproj/argo-cd/blob/9ac935d59fd90f6f568c04241047d5d45bb7bb9b/docs/operator-manual/application.yaml#L108) (in Go, rather than YAML, of course).

Once you do that, Argo CD will probably complain about the namespace being missing; you can use [this](https://github.com/argoproj/argo-cd/blob/9ac935d59fd90f6f568c04241047d5d45bb7bb9b/docs/operator-manual/application.yaml#L115) sync option in the ApplicationSpec, in order to have Argo CD create that namespace for us.

(But we should make sure to clean up those namespaces when the test ends)

Reply:
Good catch, thanks!
I'll take this in account :D